wire/msgtx.go

```diff
@@ -199,6 +199,20 @@ func (msg *MsgTx) TxSha() ShaHash {
 	return DoubleSha256SH(buf.Bytes())
 }
 
+// WTxSha generates the ShaHash of the transaction serialized according to the
+// new witness serialization defined in BIP0141. The final output is used within
+// the Segregated Witness commitment of all the witnesses within a block. If a
+// transaction has no witness data, then the witness sha, is the same as its txid.
+func (msg *MsgTx) WitnessSha() ShaHash {
```

Comment:
Can we call this `WitnessHash`?

I realize `Sha` is used in the other things and this is just being consistent, but I'd like to eventually change the funcs to be be `XHash` because I don't like that the algorithm is baked into the name since it might not always be using SHA.